### PR TITLE
Centos based image-stream for usage in OpenShift

### PR DESCRIPTION
**Pull request by sspeiche**

### CODE REVIEW

**Comment by sspeiche** [comment]
@bparees fyi

**Comment by sspeiche** [comment]
@hhorak I wasn't sure naming convention for these repos but I figured this could just be a start. I can update library and pull it into OSO starter as well. Just let me know.

**Comment by jcajka** [comment]
Could you move the stream file in to separate directory, like imagestreams or a like? @hhorak Or is there any current convention for carrying the stream files in the SCL repositories.

Otherwise LGTM. I have verifies that the stream is usable while looking in to the issue #5.

**Comment by sspeiche** [comment]
@jcajka I updated the structure to match https://github.com/sclorg/s2i-ruby-container/tree/master/imagestreams and switched the sample to use a dedicated separate repo for the example.

Will evolve the example as needed and move to sclorg (or someone can create better example)

**Comment by pvalena** [comment]
Just a side note: having some standartized locations for template and imagestream files would be really nice (I do have some automation regarding that and intend to have more).

**Comment by jcajka** [comment]
@sspeiche Thanks. If there will be agreement on some other location we can easily move it afterwards.
@pvalena What I have been looking through the SCL repos. They have the streamfiles folder. I guess you are expecting it right?

**Comment by jcajka** [comment]
@pvalena has there been any progress on setting some standard for streamfiles (location)?
I would really like to merge this soon.

**Comment by pvalena** [comment]
@jcajka the current location is fine.
Thanks!

File: imagestreams/golang-centos7.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                    "openshift.io/provider-display-name": "Red Hat, Inc.",
+                    "sampleRepo": "https://github.com/sspeiche/golang-ex.git",
+                    "supports": "golang",
+                    "tags": "builder,golang,go"
+                },
+                "from": {
+                    "kind": "ImageStreamTag",
+                    "name": "1.8"
+                },
+                "name": "latest"
```
Comment (pvalena) [inline imagestreams/golang-centos7.json]: You're missing:
```

        "referencePolicy": {
          "type": "Local"
        }
```
here. (See [example](https://github.com/sclorg/s2i-ruby-container/blob/master/imagestreams/ruby-centos7.json#L27))

File: imagestreams/golang-centos7.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                  "sampleRepo": "https://github.com/openshift/origin.git",
+                  "sampleContextDir": "examples/hello-openshift",
+                  "supports": "golang",
+                  "tags": "builder,golang,go"
+                },
+                "from": {
+                    "kind": "DockerImage",
+                    "name": "docker.io/centos/go-toolset-7-centos7:latest"
+                },
+                "name": "1.8"
```
Comment (pvalena) [inline imagestreams/golang-centos7.json]: Same here.

**Comment by pvalena** [comment]
I've not tested it (esp. the example), and there're no tests. Could you provide sample of logs for 3.9?

**Comment by sspeiche** [comment]
@pvalena I think I've made all needed changes and is ready to merge, let me know if you need anything else

File: imagestreams/golang-centos7.json [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    },
+    "spec": {
+        "tags": [
+            {
+                "annotations": {
+                    "description": "Build and run Go applications on CentOS 7. For more information about using this builder image, including OpenShift considerations, see https://github.com/sclorg/s2i-nodejs-container/blob/master/8/README.md.\n\nWARNING: By selecting this tag, your application will automatically update to use the latest version of Node.js available on OpenShift, including major versions updates.",
+                    "iconClass": "icon-go-gopher",
+                    "openshift.io/display-name": "Go (Latest)",
+                    "openshift.io/provider-display-name": "Red Hat, Inc.",
+                    "sampleRepo": "https://github.com/sspeiche/golang-ex.git",
```
Comment (pvalena) [inline imagestreams/golang-centos7.json]: Would it be possible to move sample app into https://github.com/sclorg/ org?

File: imagestreams/golang-centos7.json [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+                    "type": "Local"
+                }
+            },
+            {
+                "annotations": {
+                  "description": "Build and run Go applications on CentOS 7. For more information about using this builder image, including OpenShift considerations, see https://github.com/sclorg/s2i-nodejs-container/blob/master/8/README.md.\n\nWARNING: By selecting this tag, your application will automatically update to use the latest version of Node.js available on OpenShift, including major versions updates.",
+                  "iconClass": "icon-go-gopher",
+                  "openshift.io/display-name": "Go (1.8)",
+                  "openshift.io/provider-display-name": "Red Hat, Inc.",
+                  "sampleRepo": "https://github.com/openshift/origin.git",
```
Comment (pvalena) [inline imagestreams/golang-centos7.json]: I'm not sure about this one... Shouldn't it be the same url as above?

File: imagestreams/golang-centos7.json (hunk 3 above)
Comment (bparees) [reply imagestreams/golang-centos7.json]: @pvalena i created https://github.com/sclorg/golang-ex
feel free to update the settings/collaborators on the repo to whatever you guys want them to be.

**Comment by sspeiche** [comment]
Moved my sample to https//github.com/sclorg/golang-ex.git and updated references to it from this image-stream (in both locations).

/ptal @pvalena

**Comment by pvalena** [comment]
> before we proceed with this, we should fix the sample so that it is built using the s2i builder, not embedding its own Dockerfile.

@bparees, I somehow missed this, would you mind to elaborate? Do you mean the sample app? The `sclorg` one uses s2i, right?

**Comment by bparees** [comment]
> @bparees, I somehow missed this, would you mind to elaborate? Do you mean the sample app? The sclorg one uses s2i, right?

I don't think the sclorg one existed before now.

We just created it here:
https://github.com/sclorg/golang-ex

but yes, it now uses s2i.

**Comment by sspeiche** [comment]
@pvalena looks like this is ready to merge, not outstanding comments

**Comment by pvalena** [comment]
Although I've not tested this, it LGTM.

**Pull request merged**